### PR TITLE
Clarify documentation entry points and clean up quickstart

### PR DESCRIPTION
**Pull request by nzwulfin**
I've talked to a few folks who didn't realize we had more than the quickstart guide.  The main confusion came from the layout of the page that is listed as "Getting Started" in the nav bar, but started life as the downloads page.  

I put a bigger emphasis on the GSG and the Quickstart guide as the entry points, and cleaned up some language.  I also removed the Extreme Quickstart, that was what some folks thought were "the official instructions".

I also reflowed the Quickstart guide to put more emphasis on the steps rather than the virtualization providers to eliminate duplication.

Thoughts?

### CODE REVIEW

**Comment by jasonbrooks** [comment]
I think our quickstart process is more challenging for newcomers than it probably should be. Making your own cloud-init iso, in particular, is sort of cringeworthy --  that's got to be turning some people off. What if we led instead w/ vagrant (CentOS and Fedora each make vagrant images, and make them for both virtualbox and libvirt) or with the iso installer?

**Comment by nzwulfin** [comment]
I agree that making a cloud-init ISO or having to stand up a cloud-init source is painful, but it's also a necessary pain at this point.  Hiding that pain could make a worse impression when/if folks move to the Getting Started Guide and suddenly everything is different.  For good for worse, Atomic takes some work to get functional and we shouldn't mask that as a base for understanding the tool.

One section that was removed from the landing page was "...for the Extremely Impatient".  We can put the vagrant workflow there as a very fast, you won't really understand Atomic, but this will get you off the ground to kick the tires.  This would also mean Vagrant artifacts becoming "top level", and we'd need to get someone to own and test them. 

In short, I'm for adding Vagrant, but don't think we should mask cloud-init from users too early.

**Comment by jasonbrooks** [comment]
@nzwulfin I see that you're still referencing cockpit in `source/download/index.html.haml`, but when I run your fork of the site locally, that section doesn't seem to render? Are you seeing it locally on your end?

**Comment by jasonbrooks** [comment]
@jzb I think we should merge this and iterate from here. @nzwulfin and I just chatted on irc about next steps for filling out some additional bits, and who'll do what.

**Comment by nzwulfin** [comment]
@jzb I captured the main points of the discussion here https://gist.github.com/nzwulfin/022dc7438a8f43a999df
